### PR TITLE
Issue 509/splitting messages inbox outbox

### LINKED ISSUE #509: [Enhancement] - Migrate Message utility functions to custom hook with React Query
### Describe the Current Behavior/Feature:

With React Query in place, we could start migrating some of the utility functions we've previously built to a custom hook.

### Proposed Implementation (if applicable):

Move some of the functions from `session-helper`, `session-core`, or `MessageContext` over to a new hook called `useMessageList`.

Depending on how heavy the refactoring load would be, would likely implement this in parts.

### PR DESCRIPTION
**Pull request by leekahung**
## This PR:

Partially addresses #509 similar to PR #514. This PR is for splitting up the message pages into separate sub-routes, while retaining previous functionalities. Have updated unit tests accordingly and created separate tests for Inbox and Outbox.

## Screenshots (if applicable):

https://github.com/codeforpdx/PASS/assets/14917816/84db6895-2000-4580-b361-5f4836d69d9f

## Future Steps/PRs Needed to Finish This Work (optional):

- Updates to other Message functionalities in utils similar to what is done in PR #514, but for message building and message sending

## Issues needing discussion/feedback (optional):
**1.**    Issue #509

### CODE REVIEW

File: src/AppRoutes.jsx [hunk 1]
```diff
@@ -43,7 +45,23 @@ const AppRoutes = () => {
           <Route index element={<Contacts />} />
           <Route path=":webId" element={<Profile />} />
         </Route>
-        <Route path="/messages" element={<Messages />} />
+        <Route path="/messages">
+          {MESSAGE_PAGES_LIST.map((messagePagesProps) => (
```
Comment (milofultz) [inline src/AppRoutes.jsx]: nit: would this read nicer as `messagePage`? Then we would get `messagePage.path` and `messagePage.element`

File: src/components/Messages/MessageButtonGroup.jsx [hunk 2]
```diff
@@ -23,6 +24,14 @@ const routesArray = [{ label: 'Inbox' }, { label: 'Outbox' }];
  */
 const MessageButtonGroup = ({ setShowModal, boxType, setBoxType }) => {
   const isReallySmallScreen = useMediaQuery('(max-width: 480px)');
+  const navigate = useNavigate();
+
+  const handleNavigate = (pathLabel) => {
+    const path = pathLabel.toLowerCase();
+
+    setBoxType(path);
+    navigate(`/messages/${path}`);
```
Comment (milofultz) [inline src/components/Messages/MessageButtonGroup.jsx]: To decouple the label and the path (thinking about the future, maybe internationalization, etc.), should we add a `path` property to the `routesArray`? e.g.

```
const routesArray = [
  {
    label: 'Inbox',
    path: 'inbox',
  },
  {
    label: 'Outbox',
    path: 'outbox',
  }
];
```

Comment (milofultz) [reply src/components/Messages/MessageButtonGroup.jsx]: Then below could send `handleNavigate(item)` and destructure

Comment (milofultz) [reply src/components/Messages/MessageButtonGroup.jsx]: I understand preoptimization is the death of all code, so take this as you will 🙃

**Review by milofultz — APPROVED** [review]

File: src/AppRoutes.jsx (hunk 1 above)
Comment (leekahung) [reply src/AppRoutes.jsx]: Agreed, definitely a nicer read. I'll have this switched up.

File: src/components/Messages/MessageButtonGroup.jsx (hunk 2 above)
Comment (leekahung) [reply src/components/Messages/MessageButtonGroup.jsx]: Haha, there's definitely room for improvement. I can switch this up too.

**Pull request merged**